### DISCUSSION TITLE
Not able to change Storage path from /var/local/dcm4chee-arc/storage to /media/NAS/storage

### DISCUSSION THREAD

**Discussion by abdullah951** [dcm4che/dcm4chee-arc-light#4615] [Q&A] [answered]
When I change the docker-compose storage path from /var/local/dcm4chee-arc/storage to /opt, /other ubuntu root directory, the studies are exporting to dcm4chee server successfully. 

But when I update the storage path to /media/NAS/storage of mounted NAS, the dicom files are not exporting. Only chown.done file is created in /media/NAS/storage. LDAP is not creating fs1 folder and subfolder directories for studies. I have attached the screenshot.
![Screenshot 2024-10-30 192521](https://github.com/user-attachments/assets/ac6feea1-1d6c-4e83-a547-e117b2dd33a7)
https://github.com/dcm4che/dcm4chee-arc-light/wiki/Generic-Storage
I have tried every step listed in the generic storage.

I have tried storage.ldif way.
This is my storage.ldif file.

version: 1
dn: dcmStorageID=fs1,dicomDeviceName=dcm4chee-arc,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org
changetype: modify
replace: dcmCheckMountFilePath
dcmCheckMountFilePath: NO_MOUNT
replace: dcmDigestAlgorithm
dcmDigestAlgorithm: MD5
replace: dcmURI
dcmURI: file:///media/NAS/ihealthcure/online_storage/
replace: dcmStoragePathFormat
dcmStoragePathFormat: {now,date,yyyy/MM/dd}/{0020000D,hash}/{0020000E,hash}/{00080018,hash}

On running the following command:
ldapmodify -x -D "cn=admin,dc=dcm4che,dc=org" -W -H ldap:/// -f storage.ldif

It gives this error:
ldapmodify: wrong attributeType at line 6, entry "dcmStorageID=fs1,dicomDeviceName=dcm4chee-arc,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org"

Updated storage.ldif file:
storage.ldif file:
version: 1
dn: dcmStorageID=fs1,dicomDeviceName=dcm4chee-arc,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org
changetype: modify
replace: dcmURI
dcmURI: file:///media/NAS/ihealthcure/online_storage/

ldapsearch -x -D "cn=admin,dc=dcm4che,dc=org" -W -b "cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org" "dcmStorageID=fs1"

Result:
# extended LDIF
#
# LDAPv3
# base <cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org> with scope subtree
# filter: dcmStorageID=fs1
# requesting: ALL
#

# fs1, dcm4chee-arc, Devices, DICOM Configuration, dcm4che.org
dn: dcmStorageID=fs1,dicomDeviceName=dcm4chee-arc,cn=Devices,cn=DICOM Configur
 ation,dc=dcm4che,dc=org
dcmCheckMountFilePath: NO_MOUNT
dcmDigestAlgorithm: MD5
objectClass: dcmStorage
dcmStorageID: fs1
dcmStoragePathFormat: {now,date,yyyy/MM/dd}/{0020000D,hash}/{0020000E,hash}/{0
 0080018,hash}
dcmURI: file:///media/NAS/ihealthcure/online_storage/

# search result
search: 2
result: 0 Success

# numResponses: 2
# numEntries: 1
It replaces the storage path in LDAP configuration but still studies are not exporting using DVTk Storage SCU.

ApacheDirectoryStudio:
In ApacheDirectoryStudio, there is not parameter named dcmURI/storage.

When I create new Ldif file in ApacheDirectoryStudio  and execute it, the similar error occur like ldapmodify command above.

I have tried it with these docker-compose files.
https://github.com/dcm4che/dcm4chee-arc-light/wiki/Run-minimum-set-of-archive-services-on-a-single-host
https://github.com/dcm4che/dcm4chee-arc-light/wiki/Run-secured-archive-services-on-a-single-host
What further options do I have.
I need guidance in this regard.
Ubuntu 22.04.5

**Comment by gunterze** [comment, marked as answer]
You have to give the user with user ID: `1023` running the archive process write permission to the storage directory **at the NFS host**!

**Reply by abdullah951** [reply]
Solved with ❤ 
Thanks
